File: FunctionComparisons/function.cs
/* C# EmailValidator  */

using System;
using System.Net.Mail;

public static class EmailValidator
{
    public static (bool isValid, string message) ValidateEmail(string email)
    {
        try
        {
            var mailAddress = new MailAddress(email);
            return (true, ""); // Valid email
        }
        catch (FormatException)
        {
            return (false, "Invalid email address"); // Invalid email
        }
    }
}

class Program
{
    static void Main(string[] args)
    {
        // Input
        string emailToValidate = "test@example.com";

        // Triggering the function
        var result = EmailValidator.ValidateEmail(emailToValidate);

        // Debugging the result
        Console.WriteLine($"Validation Result: isValid={result.isValid}, message={result.message}");

        // Handling the result
        if (result.isValid)
        {
            Console.WriteLine("Email is valid!");
        }
        else
        {
            Console.WriteLine($"Error: {result.message}");
        }
    }
}

File: FunctionComparisons/function.js
const validator = require("validator");

const validateEmail = (email) => {
  if (!validator.isEmail(email)) {
    return { isValid: false, message: "Invalid email address" };
  }
  return { isValid: true };
}
// Input
const emailToValidate = "test@example.com";
// Trigger the function
const result = validateEmail(emailToValidate);
// called immediately after validating the email!
console.log("Validation Result:", result);
// separate your Validation
if (result.isValid) {
  console.log("Email is valid");
} else {
  console.log("Email is not valid");
}

const fxName = (parameter) => {
    if (!parameter) {
        console.log("this is false");
        return false;
    }
    console.log("this is true");
    return true;
}
fxName();